### PR TITLE
Add PromQL format and label matcher set/delete commands to promtool

### PR DESCRIPTION
**Pull request by robskillington**
This builds on https://github.com/prometheus/prometheus/pull/10544 which 
intended later would be added the ability to format PromQL queries to `promtool` 
to allow formatting of PromQL queries and also setting/deleting label matchers 
contained by a PromQL query (useful for programmatic manipulation of query 
definitions).

### CODE REVIEW

**Comment by robskillington** [comment]
Example usage:
```sh
$ ./promtool promql format 'foo{bar="baz"}'
foo{bar="baz"}
$ ./promtool promql format 'foo{bar="baz"}/bar'
foo{bar="baz"} / bar
$ ./promtool promql label-matchers set 'foo{bar="baz"}/bar' 'qux' 'qaz'
foo{bar="baz",qux="qaz"} / bar{qux="qaz"}
$ ./promtool promql label-matchers delete 'foo{bar="baz"}/bar' 'bar'
foo / bar
```

**Comment by robskillington** [comment]
Realized I should have brought this up directly w/ Prometheus team first, have opened the discussion here:
https://groups.google.com/g/prometheus-developers/c/toAb2wLi6Pg/m/z1_eOyO8BAAJ

**Comment by roidelapluie** [comment]
Sounds like there is consensus on the mailing list

**Comment by roidelapluie** [comment]
I have rebased this and added a `--experimental` flag

File: docs/command-line/promtool.md [hunk 1]
```diff
@@ -609,3 +611,72 @@ Create blocks of data for new recording rules.
 | rule-files | A list of one or more files containing recording rules to be backfilled. All recording rules listed in the files will be backfilled. Alerting rules are not evaluated. | Yes |
 
 
+
+
+### `promtool promql`
+
+PromQL formatting and editing. Requires the --experimental flag.
```
Comment (juliusv) [inline docs/command-line/promtool.md:618]:
```suggestion
PromQL formatting and editing. Requires the `--experimental` flag.
```

**Comment by juliusv** [comment]
:+1: otherwise

**Pull request merged**